tests/managed-package-edit-selection-test.js

```diff
@@ -0,0 +1,279 @@
+      beforeEach(function () {
+        /*
+         * The expectations in the convergent `it` blocks
+         * get run once every 10ms.  We were seeing test flakiness
+         * when a toggle action dispatched and resolved before an
+         * expectation had the chance to run.  We sidestep this by
+         * temporarily increasing the mirage server's response time
+         * to 50ms.
+         * TODO: control timing directly with Mirage
+         */
+        this.server.timing = 50;
+        return PackageEditPage.toggleIsSelected();
+      });
+
+      afterEach(function () {
+        this.server.timing = 0;
+      });
+
+      describe('clicking cancel', () => {
+        beforeEach(() => {
+          return PackageEditPage.clickCancel();
+        });
+
+        it('shows a navigation confirmation modal', () => {
+          expect(PackageEditPage.navigationModal.$root).to.exist;
+        });
+
+        describe('confirming to continue without saving', () => {
+          beforeEach(() => {
+            return PackageEditPage.navigationModal.confirmNavigation();
+          });
+
+          it('navigates from editing page', () => {
+            expect(PackageShowPage.isPresent).to.eq(true);
+          });
+
+          it('reflects the desired state of holding status', () => {
+            expect(PackageEditPage.isSelected).to.equal(false);
+          });
+        });
+        describe('confirming to keep editing', () => {
+          beforeEach(() => {
+            return PackageEditPage.navigationModal.cancelNavigation();
+          });
+
+          it('remains on the editing page', () => {
+            expect(PackageEditPage.isPresent).to.eq(true);
+          });
+        });
+      });
+
+      describe('clicking save', () => {
+        beforeEach(() => {
+          return PackageEditPage.clickSave();
+        });
+
+        it('goes to the package show page', () => {
+          expect(PackageShowPage.$root).to.exist;
+        });
+
+        it('displays the new holding status', () => {
+          expect(PackageShowPage.isVisibleToPatrons).to.equal(true);
+        });
+      });
+    });
+  });
+
+  describe('visiting the package edit page with a selected package', () => {
+    beforeEach(function () {
+      providerPackage = this.server.create('package', {
+        provider,
+        name: 'Cool Package',
+        contentType: 'E-Book',
+        isSelected: true
+      });
+      return this.visit(`/eholdings/packages/${providerPackage.id}/edit`, () => {
+        expect(PackageEditPage.$root).to.exist;
+      });
+    });
+
+    it('reflects the desired state of holding status', () => {
+      expect(PackageEditPage.isSelected).to.equal(true);
+    });
+
+    it('can toggle visibility', () => {
+      expect(PackageEditPage.isVisibleTogglePresent).to.equal(true);
+    });
+
+    it('can toggle allow kb to add titles', () => {
+      expect(PackageEditPage.hastoggleForAllowKbToAddTitles).to.equal(true);
```

Comment:
a very minor detail in terms of consistency between "isVisibleTogglePresent" and "hastoggleForAllowKbToAddTitles"